utils/maybe_async/src/lib.rs

```diff
@@ -4,67 +4,53 @@
 // LICENSE file in the root directory of this source tree.
 
 use proc_macro::TokenStream;
-use proc_macro2::TokenStream as TokenStream2;
 use quote::quote;
-use syn::{parse_macro_input, ImplItem, TraitItem};
+use syn::{parse_macro_input, Expr, ItemFn, TraitItemFn};
 
-mod parse;
-use parse::Item;
-
-mod visit;
-use visit::AsyncAwaitRemoval;
-
-/// maybe_async attribute macro
+/// maybe_async procedural attribute macro
+///
+/// Parses a function (regular or trait) and conditionally adds the `async` keyword
+/// depending on the `async` feature flag being enabled.
 #[proc_macro_attribute]
-pub fn maybe_async(_args: TokenStream, input: TokenStream) -> TokenStream {
-    let mut item = parse_macro_input!(input as Item);
-
-    let token = if cfg!(feature = "async") {
-        convert_async(&mut item)
-    } else {
-        convert_sync(&mut item)
-    };
-
-    token.into()
-}
-
-fn convert_sync(input: &mut Item) -> TokenStream2 {
-    match input {
-        Item::Impl(item) => {
-            for inner in &mut item.items {
-                if let ImplItem::Fn(ref mut method) = inner {
-                    if method.sig.asyncness.is_some() {
-                        method.sig.asyncness = None;
-                    }
-                }
+pub fn maybe_async(_attr: TokenStream, input: TokenStream) -> TokenStream {
```

Comment:
I would get rid of line 10 and add an example (same as in readme):

```Rust
/// Parses a function (regular or trait) and conditionally adds the `async` keyword
/// depending on the `async` feature flag being enabled.
///
/// For example:
/// ```ignore
/// trait ExampleTrait {
///     #[maybe_async]
///     fn say_hello(&self);
/// 
///     #[maybe_async]
///     fn get_hello(&self) -> String;
/// }
///
/// 
/// #[maybe_async]
/// fn hello_world() {
///     // ...
/// }
/// ```
/// When the `async` feature is enabled, will be transformed into:
/// ```ignore
/// trait ExampleTrait {
///     async fn say_hello(&self);
/// 
///     async fn get_hello(&self) -> String;
/// }
///
/// 
/// async fn hello_world() {
///     // ...
/// }
/// ```
```